pkg/operator/ceph/spec/label.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	"github.com/rook/rook/pkg/operator/ceph/version"
+	apps "k8s.io/api/apps/v1"
+	batch "k8s.io/api/batch/v1"
+)
+
+const (
+	// CephVersionLabelKey is the key used for reporting the Ceph version which Rook has detected is
+	// configured for the labeled resource.
+	CephVersionLabelKey = "ceph-version"
+)
+
+// Add the Ceph version to the given labels.  This should *not* be used on pod specifications,
+// because this will result in the deployment/daemonset/ect. recreating all of its pods even if an
+// update wouldn't otherwise be required. Upgrading unnecessarily increases risk for loss of data
+// reliability, even if only briefly.
+func addCephVersionLabel(cephVersion version.CephVersion, labels map[string]string) {
+	labels[CephVersionLabelKey] = cephVersion.String()
+}
+
+// AddCephVersionLabelToDeployment adds a label reporting the Ceph version which Rook has detected is
+// running in the Deployment's pods.
+func AddCephVersionLabelToDeployment(cephVersion version.CephVersion, d *apps.Deployment) {
```

Comment:
What if the method took the `v1.ObjectMeta` type to which the labels would be applied? This way you could pass `deployment.ObjectMeta` (or similar) and you'd just need a single helper method.

Reply:
Hmm ... I didn't think about that, and I like it. But I think it's slightly safer to keep it this way so that no one down the road is tempted to use the method for making updates to pod specs within the controllers, which would cause unnecessary updates.